src/memspaces/memspace_numa.c

```diff
@@ -0,0 +1,62 @@
+/*
+ *
+ * Copyright (C) 2023 Intel Corporation
+ *
+ * Under the Apache License v2.0 with LLVM Exceptions. See LICENSE.TXT.
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+ *
+ */
+
+#include <stdlib.h>
+
+#include "../memory_targets/memory_target_numa.h"
+#include "../memspace_internal.h"
+#include "memspace_numa.h"
+
+enum umf_result_t
+umfMemspaceCreateFromNumaArray(size_t *nodeIds, size_t numIds,
```

Comment:
why not just create a memspace from the array of memory targets?

Reply:
This is just a wrapper that does exactly that underneath. It will be used for implementing HOST_MEMSPACE.

Also, this function was initially public (and we don't expose memory_targets in the public API for now) You suggested to hide this function for now which I did. However, we do plan to expose this function in future (possibly before we expose memory_targets).